frontend/src/employee-frontend/components/messages/MessageEditor.tsx

```diff
@@ -187,20 +195,33 @@ export default React.memo(function MessageEditor({
   )
   useEffect(
     function updateReceiversOnSenderChange() {
+      const previousReceivers = message.recipientIds
       const nestedAccount = nestedAccounts.find(
         (account) => account.account.id === message.sender.value
       )
       if (nestedAccount && !isNestedGroupMessageAccount(nestedAccount)) {
-        setReceiverTree(availableReceivers)
+        setReceiverTree(
```

Comment:
Moreover, this if-else-if logic could be simplified by returning early or throwing `if (!nestedAccount)`.

Reply:
Thanks. Refactored as suggested.